### PR TITLE
Support schema referencing in media type object

### PR DESCRIPTION
**Pull request by kylef**

### CODE REVIEW

File: packages/fury-adapter-oas3-parser/lib/parser/oas/parseComponentsObject.js [hunk 1]
```diff
@@ -31,9 +31,17 @@ function parseComponentsObject(context, element) {
   const validateIsObject = key => R.unless(isObject,
     createWarning(namespace, `'${name}' '${key}' is not an object`));
 
+  const parseSchemaMember = member => pipeParseResult(namespace,
+    R.compose(parseSchemaObject(context), getValue),
+    (dataStructure) => {
+      // eslint-disable-next-line no-param-reassign
```
Comment (kylef) [inline packages/fury-adapter-oas3-parser/lib/parser/oas/parseComponentsObject.js]: I understand what this eslint rule is meant to protect against and in a lot of cases I agree with it. However, in this particular line I think this makes sense, we're modifying the dataStructure to attach the ID.

File: packages/fury-adapter-oas3-parser/lib/parser/oas/parseComponentsObject.js [hunk 2]
```diff
@@ -31,9 +31,17 @@ function parseComponentsObject(context, element) {
...
+      dataStructure.content.id = member.key.clone();
```
Comment (kylef) [inline packages/fury-adapter-oas3-parser/lib/parser/oas/parseComponentsObject.js]: In https://github.com/apiaryio/api-elements.js/pull/86 I moved this logic a bit too far away and into parseOpenAPI and I think instead it should be here inside the components parser so the components contain the ID used for referencing.

File: packages/fury-adapter-oas3-parser/lib/parser/parseReference.js [hunk 3]
```diff
@@ -8,7 +8,19 @@ function isReferenceObject(element) {
 
 function parseReference(component, parser, context, element) {
   if (isReferenceObject(element)) {
-    return parseReferenceObject(context, component, element);
+    const result = parseReferenceObject(context, component, element);
```
Comment (honzajavorek) [inline packages/fury-adapter-oas3-parser/lib/parser/parseReference.js]: Took me a moment to realize that this isn't a smelly variable name, but a variable for "parse result". Perhaps `parseResult` would cognitively lead better to realizing what's in it?

Comment (kylef) [reply packages/fury-adapter-oas3-parser/lib/parser/parseReference.js]: I'll prepare a separate PR changing the variable name, this is a frequent pattern in most files. I'll sed the code base and do it all.

**Pull request merged**